openedx/adg/lms/applications/handlers.py

```diff
@@ -16,16 +16,19 @@ def update_business_line_group(instance, **kwargs):  # pylint: disable=unused-ar
     Modify user group on updating a Business Line
     """
     if instance.pk is None:
-        Group.objects.create(name=instance.title)
+        group = Group.objects.create(name=instance.title)
+        instance.group = group
     else:
         old_instance = BusinessLine.objects.get(pk=instance.pk)
         if old_instance.title != instance.title:
-            Group.objects.filter(name=old_instance.title).update(name=instance.title)
+            instance.group.name = instance.title
+            instance.group.save()
 
 
 @receiver(post_delete, sender=BusinessLine)
 def delete_business_line_group(instance, **kwargs):  # pylint: disable=unused-argument
     """
     Delete user group on deleting a Business Line
     """
+    # TODO: Use instance.group.delete(), once LP-2479 is deployed on prod.
```

Comment:
We won't have any option to delete the existing BUs, because existing BUs doesn't have a group associated with a FK. Ideally we should delete existing BUs before deploying this change the server, but this is just to support any case where we would miss to do so.

Reply:
https://github.com/OmnipreneurshipAcademy/edx-platform/pull/166/files#r580025455